pkg/pillar/cmd/tpmmgr/tpmmgr.go

```diff
@@ -999,6 +1000,63 @@ func createEcdhCert() error {
 	return nil
 }
 
+func publishAttestCert(ctx *tpmMgrContext, config types.AttestCert) {
+	key := config.Key()
+	log.Debugf("publishAttestCert %s\n", key)
+	pub := ctx.pubAttestCert
+	pub.Publish(key, config)
+	log.Debugf("publishAttestCert %s Done\n", key)
+}
+
+func getECDHCert(certPath string) ([]byte, error) {
+	certBytes, err := ioutil.ReadFile(certPath)
+	if err != nil {
+		errStr := fmt.Sprintf("getECDHCert failed while reading ECDH certificate: %v",
+			err)
+		log.Error(errStr)
+		return []byte{}, errors.New(errStr)
+	}
+	return certBytes, nil
+}
+
+func getCertHash(cert []byte, hashAlgo types.CertHashType) ([]byte, error) {
+	certHash := sha256.Sum256(cert)
+	switch hashAlgo {
+	case types.CertHashTypeSha256First16:
+		return certHash[:16], nil
+	default:
+		return []byte{}, fmt.Errorf("Unsupported cert hash type")
```

Comment:
How about adding a %d to include the hashAlgo in the error message?

Reply:
Yeah, it makes sense. I will add the hashAlgo in the error message.